### DISCUSSION TITLE
When do we need Messenger?

### DISCUSSION THREAD

**Discussion by karaka200** [symfony/symfony#45293] [Q&A] [answered]
Is it better for all sites to use Messenger (`symfony/messenger`) or not just for large sites that do complex tasks? I did not understand this from it documents (https://symfony.com/doc/current/messenger.html)

**Comment by GromNaN** [comment, marked as answer]
Symfony Messenger is useful to dissociate a task from the main process. Examples:
- send an email when a user login
- invalidate cache when an article is published
- index to a search service when a document is updated

I found this page that describe use-cases of messages.  https://www.cloudamqp.com/blog/rabbitmq-use-cases-explaining-message-queues-and-when-to-use-them.html

**Reply by karaka200** [reply]
Thank you, it was very helpful
